easybuild/main.py

```diff
@@ -64,6 +64,12 @@
 from easybuild.tools.testing import create_test_report, overall_test_report, regtest, session_module_list, session_state
 from easybuild.tools.version import this_is_easybuild
 
+try:
+    from humanfriendly.terminal import terminal_supports_colors
+except ImportError:
+    # provide an approximation that should work in most cases
```

Comment:
(Kenneth Hoste, Fri, Mar 25, 2016 at 03:47:26PM -0700:)
> > @@ -64,6 +64,12 @@
> >  from easybuild.tools.testing import create_test_report, overall_test_report, regtest, session_module_list, session_state
> >  from easybuild.tools.version import this_is_easybuild
> >
> > +try:
> > +    from humanfriendly.terminal import terminal_supports_colors
> > +except ImportError:
> > +    # provide an approximation that should work in most cases
>
> so, when would it not work?

`os.isatty()` tells you if the stream is connected to a terminal, it
does not say whether the terminal supports ANSI colors.

I guess it's OK to presume that everyone nowadays is using a terminal
emulator with ANSI color support (hence "most cases") but you never
know...  (I have seen people still run FVWM as their window manager in
2015; I would not be surprised to hear that somebody is stil using a
pre-ANSI `xterm` ...)


> do we really needed `humanfriendly` at all?

It's a dependency of `coloredlogs` (the Python package), so if you are
going to have colored logs, `humanfriendly` is going to be there anyway.